coderd/tailnet.go

```diff
@@ -200,7 +201,24 @@ type ServerTailnet struct {
 	transport *http.Transport
 }
 
+// insureTLSConfig returns a tls config that does not verify
+// the server's certificate chain.
+func insecureTLSConfig() *tls.Config {
+	return &tls.Config{
+		MinVersion:         tls.VersionTLS12,
+		InsecureSkipVerify: true,
+	}
+}
+
 func (s *ServerTailnet) ReverseProxy(targetURL, dashboardURL *url.URL, agentID uuid.UUID) (_ *httputil.ReverseProxy, release func(), _ error) {
+	transport := s.transport
+
+	// We don't verify certificates for localhost applications.
+	if targetURL.Scheme == "https" {
+		transport = transport.Clone()
+		transport.TLSClientConfig = insecureTLSConfig()
+	}
```

Comment:
might as well just set this in the `init()` function that creates the transport.